### PR TITLE
fix: overview search

### PR DESCRIPTION
**Pull request by crazyair**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 713da20</samp>

Improved theme component for component overview. Refactored imports and query selector in `.dumi/theme/builtins/ComponentOverview/index.tsx` to follow best practices and avoid potential errors.

### <samp>🤖 Generated by Copilot at 713da20</samp>

*  Reorder and group imports by external and internal modules for readability and consistency ([link](https://github.com/ant-design/ant-design/pull/44523/files?diff=unified&w=0#diff-f1df5cf9560d26c9862a4824e7d1a13f6a8bec1ec7338e9327f3e7a6a74c1585L3-R58))
*  Use generated class name from `useStyle` hook instead of hard-coded string for card element query selector, to avoid conflicts and improve maintainability ([link](https://github.com/ant-design/ant-design/pull/44523/files?diff=unified&w=0#diff-f1df5cf9560d26c9862a4824e7d1a13f6a8bec1ec7338e9327f3e7a6a74c1585L110-R114))

### CODE REVIEW

File: .dumi/theme/builtins/ComponentOverview/index.tsx [hunk 1]
```diff
@@ -107,7 +111,7 @@ const Overview: React.FC = () => {
 
   const onKeyDown: React.KeyboardEventHandler<HTMLInputElement> = (event) => {
     if (event.keyCode === 13 && search.trim().length) {
-      sectionRef.current?.querySelector<HTMLElement>('.components-overview-card')?.click();
+      sectionRef.current?.querySelector<HTMLElement>(`.${styles.componentsOverviewCard}`)?.click();
```
Comment (MadCcc) [inline .dumi/theme/builtins/ComponentOverview/index.tsx:114]: 这个实现居然这么黑？直接调用 click 事件的 handler 是不是就可以了

Comment (crazyair) [reply .dumi/theme/builtins/ComponentOverview/index.tsx:114]: 之前豆酱偷懒的吧，其实用搜索结果的第一个执行 history.push 也可以

Comment (crazyair) [reply .dumi/theme/builtins/ComponentOverview/index.tsx:114]: > 这个实现居然这么黑？直接调用 click 事件的 handler 是不是就可以了

没 click 吧，是`<a href="xxx"`

Comment (zombieJ) [reply .dumi/theme/builtins/ComponentOverview/index.tsx:114]: 😈 我不是，我没有，别瞎说。我只是大自然的搬运工~

https://github.com/ant-design/ant-design/blame/4.x-stable/site/theme/template/Content/ComponentOverview.jsx#L54

Comment (crazyair) [reply .dumi/theme/builtins/ComponentOverview/index.tsx:114]: 这玩意没单测，要不搬运的时候就能发现了

Comment (zombieJ) [reply .dumi/theme/builtins/ComponentOverview/index.tsx:114]: 这里是有 data 的，直接从 data 里取出来，就不要再调 dom 了

**Review by afc163 — APPROVED** [review]

**Pull request merged**